streams/jvm/src/main/scala/zio/stream/compression/Gunzipper.scala

```diff
@@ -0,0 +1,184 @@
+package zio.stream.compression
+
+import java.util.zip.{ CRC32, DataFormatException, Inflater }
+import java.{ util => ju }
+
+import scala.annotation.tailrec
+
+import zio._
+
+private[compression] class Gunzipper private (var state: Gunzipper.State) {
```

Comment:
I suggest making the functionality in this file mutable and ZIO-less. It'd be much faster this way and you're only doing synchronous effects and error handling, so you can just replace that with exceptions.